packages/common/src/core/schemas/internal/metrics/MetricSchema.ts

```diff
@@ -66,6 +66,10 @@ export const MetricSchema: IConfigurationSchema = {
     },
     sourceLink: {
       type: "string",
+      if: { minLength: 1 },
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
I know you're just following the existing pattern, but I don't love that we'll need to include this `ts-ignore` every time we want to use the hub-defined `url` format. I'm wondering if we can somehow extend our `IConfigurationSchema` (which extends the `JSONSchema`) to allow this format? 

Anyways, that's outside the scope of this issue, but may be something we want to consider.